cmd/lifecycle/detector.go

```diff
@@ -79,11 +79,15 @@ func (da detectArgs) detect() (lifecycle.BuildpackGroup, lifecycle.BuildPlan, er
 		Logger:        cmd.Logger,
 	})
 	if err != nil {
-		if err == lifecycle.ErrFail {
+		if err == lifecycle.ErrFail || err == lifecycle.ErrFailWithErrors {
 			cmd.Logger.Error("No buildpack groups passed detection.")
 			cmd.Logger.Error("Please check that you are running against the correct path.")
 		}
-		return lifecycle.BuildpackGroup{}, lifecycle.BuildPlan{}, cmd.FailErrCode(err, cmd.CodeFailedDetect, "detect")
+		exitCode := cmd.CodeFailedDetect
```

Comment:
If we do this we are using `CodeFailedDetect` for more than just ("sucessfully" failed detection). The lifecycle itself can throw errors. I think we need a third error code to separate these from "nothing detected".

I think we should do a switch on `err` and have a `default` in addition to `ErrFail` and `ErrFailWIthErrors` cases